### ISSUE TITLE
External provider login redirects to main site.

### ISSUE THREAD

**Issue by tech4him1** [netlify/netlify-identity-widget#86]
If I am working on a `localhost` site, and I try to login with an external OAuth provider, I am redirected back to the main site (the one with Identity), instead of my `localhost` dev site.

**Comment by KrishnaPG** [comment]
Is there anyway (perhaps on the Netlify identity settings UI) to specify the redirection URL?

**Comment by marcobiedermann** [comment]
Same here. I setup my site using Gatsby and netlify-cms. I deployed everything to netlify and enabled identity services. When trying this locally using "http://localhost:9000" in the site URL from admin prompt I get this:

![](https://user-images.githubusercontent.com/5244986/34179877-eaae6b02-e50c-11e7-852a-cec7c021fc50.png)

Also tried to copy my remote settings file locally but this does not seam to work either.

**Comment by tech4him1** [comment]
@marcobiedermann No, that is a different issue. When it asks you for the site URL, you need to enter the actual deployed site URL, so that it can access your Identity instance (your users, etc.).

**Comment by marcobiedermann** [comment]
@tech4him1 I thought so but after entering my live site URL I get redirected to the live site instead of localhost (which is the right behaviour in my opinion).
Still I don't know how to enter the cms backend panel from localhost.

**Comment by tech4him1** [comment]
@marcobiedermann The redirection bug **is** what this issue is about. The bug only happens when you try to log in with an external provider (GitHub, Google, etc.). If you log in with a email/password it should not redirect. If you need any more help setting it up you can ask in the Gitter: https://gitter.im/netlify/

**Comment by erquhart** [comment]
@bcomnes can you provide a status update on this?

**Comment by bcomnes** [comment]
We just started work on this, so hopefully soon!

**Comment by tech4him1** [comment]
@bcomnes If I understand correctly, https://github.com/netlify/gotrue/pull/132 fixed this, but it doesn't seem to be deployed on Netlify's instance yet?

**Comment by bcomnes** [comment]
We have a fix in the code, I don't have an ETA yet but I'm eager to get it deployed.

**Comment by jsamr** [comment]
@bcomnes any updates ?

**Comment by bcomnes** [comment]
I confirmed this today as correct behavior that isn't changing soon.  

What we can do in the meantime:

- Provider logins won't work in devmode (we either disable them or include devmode only messaging that they will redirect to the live site)
- Email/password login is always enabled in devmode, because its the only method that does not require a redirect.  When we support provider only logins, the email/password login will only show in devmode, so you can log in.

See https://github.com/netlify/netlify-identity-widget/issues/100

I would, one day, like to see provider logins work in devmode, so I'm leaving this open until we make a final decision on the matter.

**Comment by laurajodz** [comment]
FYI Devex owns this repo but doesn't use GitHub project boards like Eng does. So there is no project to add to.

(https://netlify.slack.com/archives/CCC1HDWQY/p1600806294044200)

**Comment by signalkuppe** [comment]
+1

**Comment by andrewgremlich** [comment]
How's the progress on this? It seems like it still exists with the app that I'm working on.

**Comment by contendaClara** [comment]
any updates? I get this behavior though I don't use an external OAuth provider

**Comment by khylias** [comment]
It is still not possible to authenticate on a development environment with this widget. After connecting to the provider, for example Github, we are always redirected to the production url. But it is on the development environment that we need to be redirected to get the access_token.

This has nothing to do with Netlify CMS. 

Do you intend to investigate the problem?

**Comment by filipesperandio** [comment]
It is a little sad that 7 years later and this is still an issue.

Using DecapCMS, if we flip the git-gateway option + netlify-identity to use the github backend instead, local dev works like a charm. I believe the auth process just uses a window popup instead so it can avoid going to the prod website.
We could use a similar approach when using the external providers for local dev to also work.
